routes/cart.router.js

```diff
@@ -0,0 +1,198 @@
+const router = require("express").Router();
+// const User = require("../models/user.model");
+const { Book } = require("../models/book.model");
+const Cart = require("../models/cart.model");
+
+const getFormatDetails = (book, currentFormat) => {
+  const currentFormatIndex = book.formats.findIndex(
+    (format) => format.type === currentFormat
+  );
+  // return book.formats[currentFormatIndex];
+  return {
+    price: +book.formats[currentFormatIndex].price.toFixed(2),
+    discount: +(
+      (book.formats[currentFormatIndex].discount / 100) *
+      book.formats[currentFormatIndex].price
+    ).toFixed(2),
+    stock: book.formats[currentFormatIndex].stock,
+  };
+};
+
+router.param("cartId", async (req, res, next, cartId) => {
+  try {
+    const cartFound = await Cart.findById(cartId).populate("items.book");
+    if (!cartFound) {
+      res.status(404).json({ success: false, message: "Cart not found!" });
+    }
+    req.cart = cartFound;
+    next();
+  } catch (error) {
+    console.error(error);
+    res
+      .status(+error.code)
+      .json({ success: false, message: "Error while retrieving cart." });
+  }
+});
+
+router
+  .route("/:cartId")
+  .get(async (req, res) => {
+    try {
+      res.status(200).json({ success: true, cart: req.cart });
+    } catch (error) {
+      console.error(error);
+      res
+        .status(+error.code)
+        .json({ success: false, message: "Cart not found in database." });
+    }
+  })
+  .post(async (req, res) => {
```

Comment:
You can have separate controller folder which would include all controlling function
for example
In `routes/cart.routes.js`
`router.get("/:userID", getUserCart)`

In `controllers/cart.controllers.js`
`const getUserCart = async(req, res) => {}`

separating will make code more readable.

you can read more about it here  [https://lo-victoria.com/build-a-rest-api-with-nodejs-routes-and-controllers](https://lo-victoria.com/build-a-rest-api-with-nodejs-routes-and-controllers)